### DISCUSSION TITLE
Script not working when entering website

### DISCUSSION THREAD

**Discussion by CHolitas** [swup/swup#393] [Help]
Ok, so I have a script that if I enter the website it works properly but if I enter another page in my website and re enter this one the script doesn't work. Can someone help me?

PS: Here is the script, it's a simple typing effect
`<script>
        var TxtType = function (el, toRotate, period) {
            this.toRotate = toRotate;
            this.el = el;
            this.loopNum = 0;
            this.period = parseInt(period, 10) || 2000;
            this.txt = '';
            this.tick();
            this.isDeleting = false;
        };

        TxtType.prototype.tick = function () {
            var i = this.loopNum % this.toRotate.length;
            var fullTxt = this.toRotate[i];

            if (this.isDeleting) {
                this.txt = fullTxt.substring(0, this.txt.length - 1);
            } else {
                this.txt = fullTxt.substring(0, this.txt.length + 1);
            }

            this.el.innerHTML = '<span class="wrap">' + this.txt + '</span>';

            var that = this;
            var delta = 200 - Math.random() * 100;

            if (this.isDeleting) {
                delta /= 2;
            }

            if (!this.isDeleting && this.txt === fullTxt) {
                delta = this.period;
                this.isDeleting = true;
            } else if (this.isDeleting && this.txt === '') {
                this.isDeleting = false;
                this.loopNum++;
                delta = 500;
            }

            setTimeout(function () {
                that.tick();
            }, delta);
        };

        window.onload = function () {
            var elements = document.getElementsByClassName('typewrite');
            for (var i = 0; i < elements.length; i++) {
                var toRotate = elements[i].getAttribute('data-type');
                var period = elements[i].getAttribute('data-period');
                if (toRotate) {
                    new TxtType(elements[i], JSON.parse(toRotate), period);
                }
            }
            // INJECT CSS
            var css = document.createElement("style");
            css.type = "text/css";
            css.innerHTML = ".typewrite > .wrap { border-right: 0.08em solid #fff}";
            document.body.appendChild(css);
        };
    </script>`

**Comment by AugustMiller** [comment]
Please review the [Reloading Javascript](https://swup.js.org/getting-started/reloading-javascript) page of the documentation.

Swup requires that you think a little differently about the page's lifecycle, and won't automatically work with scripts that make use of `window.onload`, `DOMContentLoaded`, etc!

**Comment by dangelion** [comment]
Hi @AugustMiller 
just read your comment about `DOMContentLoaded` etc. I've this strange (and random) problem:
https://github.com/swup/swup/issues/395

You make me think it could be the origin of my issue too. I'm just using this structure:
https://github.com/swup/swup/issues/358#issuecomment-833309664
that makes use of `document.readyState === 'complete'`, `DOMContentLoaded` and others.
Is it correct? Could you help me?
Thanks

**Comment by possible-is** [comment]
Hi, i would like to chime in here and ask a question related to that:
I have successfully made use of the Reloading Javascript page and have now a section in footer of the page that loads and unloads scripts related to certain pages via swup.on('contentReplaced', init);.
But: Since there is always the chance of people entering the page via URLs, they might enter any page without the transitions taking place yet. This means, i need to have my scripts loaded by Swup after transitions, and additionally have them included in each respective page onload.
This creates redundancies that i would like to avoid. Is there a way to expand swup.on('contentReplaced', init); to something that works with contentReplaced and DOMContentLoaded as well?
This way i could bundle all scripts in just one place.
